src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/OutputProducer.cs

```diff
@@ -42,7 +43,83 @@ public OutputProducer(IPipeWriter pipe, Frame frame, string connectionId, IKestr
             _flushCompleted = OnFlushCompleted;
         }
 
-        public Task WriteAsync(
+        public void Write(ArraySegment<byte> buffer, bool chunk = false)
+        {
+            WriteAsync(buffer, default(CancellationToken), chunk).GetAwaiter().GetResult();
+        }
+
+        public Task WriteAsync(ArraySegment<byte> buffer, bool chunk = false, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                _frame.Abort(error: null);
```

Comment:
I understand this was here already, but it doesn't seem right to abort the entire thing because a write is cancelled. Since this is under user control, someone could be using a ct to decide not to write at a given point if the token is cancelled, but would still want to continue running the app and doing further writes. TBH I don't know if this is realistic, but it's easily doable.